nncf/quantization/algorithms/fast_bias_correction/torch_backend.py

```diff
@@ -0,0 +1,137 @@
+        return PT_OPERATOR_METATYPES
+
+    @property
+    def tensor_processor(self) -> PTNNCFCollectorTensorProcessor:
+        return PTNNCFCollectorTensorProcessor()
+
+    @staticmethod
+    def target_point(target_type: TargetType, target_node_name: str, port_id: int) -> PTTargetPoint:
+        if NNCFGraphNodeType.INPUT_NODE in target_node_name or target_type == TargetType.POST_LAYER_OPERATION:
+            port_id = None
+        if target_type in PTFastBiasCorrectionAlgoBackend.TARGET_TYPE_TO_PT_INS_TYPE_MAP:
+            target_type = PTFastBiasCorrectionAlgoBackend.TARGET_TYPE_TO_PT_INS_TYPE_MAP[target_type]
+        return PTTargetPoint(target_type, target_node_name, input_port_id=port_id)
+
+    @staticmethod
+    def create_bias_correction_command(node: NNCFNode,
+                                       bias_value: np.ndarray,
+                                       nncf_graph: NNCFGraph) -> PTBiasCorrectionCommand:
+        return create_bias_correction_command(node, bias_value)
+
+    @staticmethod
+    def model_extraction_command(inputs: List[str], outputs: List[str]) -> PTModelExtractionCommand:
+        return PTModelExtractionCommand(inputs[0])
+
+    @staticmethod
+    def mean_statistic_collector(reduction_shape: ReductionShape,
+                                 inplace: bool,
+                                 num_samples: Optional[int] = None,
+                                 window_size: Optional[int] = None) -> PTMeanStatisticCollector:
+        return PTMeanStatisticCollector(reduction_shape, num_samples, window_size)
+
+    @staticmethod
+    def get_sub_input_output_names(subgraph: NNCFNetwork) -> Tuple[str, str]:
+        # Pytorch does not have name for extracted node
+        return None, None
+
+    @staticmethod
+    def create_input_data(
+        shape: Tuple[int], data: List[torch.Tensor], input_name: str, channel_axis: int
+    ) -> torch.Tensor:
+        blob = torch.zeros(shape, dtype=data[0].dtype)
+        for j, idx in enumerate(np.ndindex(blob.shape[channel_axis])):
+            index = tuple(slice(None) if i != channel_axis else idx for i in range(blob.ndim))
+            blob[index] = data[j]
+        return blob
+
+    @staticmethod
+    def get_bias_value(node: NNCFNode, nncf_graph: NNCFGraph, model: NNCFNetwork) -> np.ndarray:
+        return get_bias_value(node, model)
+
+    @staticmethod
+    def get_activation_port_ids_for_bias_node(node: NNCFNode) -> Tuple[int, int]:
+        return 0, 0
+
+    @staticmethod
+    def process_model_output(raw_data: Dict, output_name: str) -> PTNNCFTensor:
+        return PTNNCFTensor(raw_data)
+
+    @staticmethod
+    def is_quantized_weights(node: NNCFNode, nncf_graph: NNCFGraph, model: NNCFNetwork) -> bool:
+        node_module = model.get_containing_module(node.node_name)
+        for pre_op in node_module.pre_ops.values():
+            if isinstance(pre_op, UpdateWeight) and isinstance(pre_op.op, (BaseQuantizer, FakeQuantize)):
```

Comment:
Why there are no checks for quantized weights? Can quantizers be followed by the `Reshape` or other types of layers before weighted layers?

Reply:
> Why there are no checks for quantized weights? 

`if isinstance(pre_op, UpdateWeight) and isinstance(pre_op.op, (BaseQuantizer, FakeQuantize)):` is check for quantized weights. If Conv module have `UpdateWeight` operator that run FQ for weights, it means that node is quantized.
Or what about this question?

> Can quantizers be followed by the Reshape or other types of layers before weighted layers?

No, in torch FQ add in preops for weight directly.
It's limitation of PT quantization, that supported only weights that collected in `weights` attribute of `conv` module (only default conv module from torch). If weights is output of another node NNCF will no quantize this module.

Reply:
`UpdateWeight` may be used for `outputs insertion` as we found previously. So, there is no guarantee that `UpdateWeight` means `weights quantization`, especially when we check all outputs, instead of the `weighted` port.

Reply:
No. `UpdateWeight` can update only weight, and if FQ is `op` of `UpdateWeight` it's means that `weights quantization`
`UpdateWeight` used for add NNCF operation (FQ, prunning by mask) for work with weights.

`outputs insertion` name of function was incorrected and changed it to `_apply_insertion_transformations`.

And again in torch no output/input nodes in graph, and it's not a graph.

Reply:
`UpdateWeight` used for add NNCF operation (FQ, pruning by mask) for work with weights.

What should be detailed in this answer to answer your question?